async-http-client-backend/zio1/src/test/scala/sttp/client3/asynchttpclient/zio/SttpBackendStubZioTests.scala

```diff
@@ -73,4 +74,45 @@ class SttpBackendStubZioTests extends AnyFlatSpec with Matchers with ScalaFuture
 
     runtime.unsafeRun(effect).map(_.body).toList shouldBe List(Right("a"), Right("b"), Right("c"), Right("a"))
   }
+
+  it should "lift errors due to mapping with impure functions into the response monad" in {
+    val backend: SttpBackendStub[Task, Any] =
+      SttpBackendStub(new RIOMonadAsyncError[Any]).whenAnyRequest.thenRespondOk()
+
+    val error = new IllegalStateException("boom")
+
+    val r = basicRequest
+      .post(uri"http://example.org")
+      .response(asStringAlways.map[Int](_ => throw error))
+      .send(backend)
+
+    runtime.unsafeRun(r.either) match {
+      case Left(_: IllegalStateException) => succeed
+      case _                              => fail(s"Should be a failure: $r")
+    }
+  }
+
+  trait TestStreams extends Streams[TestStreams] {
```

Comment:
there is already TestStreams in SttpBackendStubTests maybe import is enought ?

Reply:
not possible because it's defined as an inner class